Library/Homebrew/dev-cmd/vendor-gems.rb

```diff
@@ -59,12 +59,16 @@ def vendor_gems
           mechanize
         ].each do |gem|
           ohai "gem install #{gem}"
-          safe_system "gem", "install", "mechanize", "--install-dir", "vendor",
+          safe_system "gem", "install", gem, "--install-dir", "vendor",
                       "--no-document", "--no-wrappers", "--ignore-dependencies", "--force"
           (HOMEBREW_LIBRARY_PATH/"vendor/gems").cd do
-            if (source = Pathname.glob("#{gem}-*/").first)
-              FileUtils.ln_sf source, gem
-            end
+            source = Pathname.glob("#{gem}-*/").first
+            next if source.blank?
+
+            target = Pathname.new(gem)
+            next if target.exist? && target.symlink? && (target.realpath == source.realpath)
+
+            FileUtils.ln_sf source, gem
```

Comment:
In the case of a version update, we probably need to remove the existing symlink first to avoid a similar issue happening.

Reply:
I always thought that the existing symlink is broken in the case of version updates, but I guess I'm not actually certain about that.

Let's just always remove the symlink, which is probably the intention of `ln_sf`.

Reply:
> I always thought that the existing symlink is broken in the case of version updates, but I guess I'm not actually certain about that.

In theory it should, but seems safer to not rely on it when there's an simple way to not to.